src/Location/WorkingSchedule.php

```diff
@@ -408,10 +408,10 @@ protected function applyTimezone(DateTimeInterface $date)
         return $date;
     }
 
-    protected function filterTimeslot(Collection $timeslot, DateTime $checkDateTime)
+    protected function filterTimeslot(Collection $timeslot, DateTime $checkDateTime, int $leadTime)
     {
-        return $timeslot->filter(function (DateTime $dateTime) use ($checkDateTime) {
-            return Carbon::instance($checkDateTime)->lte($dateTime);
+        return $timeslot->filter(function (DateTime $dateTime) use ($checkDateTime, $leadTime) {
+            return Carbon::now()->diffInMinutes($dateTime) > $leadTime AND Carbon::instance($checkDateTime)->lte($dateTime);
```

Comment:
Should probably split this to avoid side-effects. First filter out slots that are less than the checkDateTime before the diff check.

Reply:
Changed :)

Reply:
Looking neat! One last thing, pass `$dateTime` instead and remove `$dateTime->copy()`

Reply:
Done! :)